modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+        drupal_render($form[$key]['links']),
+      ),
+    );
+
+    $rows[] = $row;
+  }
+
+  $output = theme(
+    'table',
+    array(
+      'header' => $header,
+      'rows' => $rows,
+      'attributes' => array(
+        'id' => 'ding_nodelist-links-table',
+      ),
+    )
+  );
+  $output .= drupal_render_children($form);
+  return $output;
+}
+
+/**
+ * Theme function to render the backend list of templates.
+ */
+function theme__ding_nodelist_templates_table($variables) {
+  $header = array(
+    t('Machine name'),
+    t('Title'),
+    t('Content type'),
+    t('List type'),
+    t('Visible'),
+  );
+  $rows = array();
+  $form = $variables['form'];
+  foreach (element_children($form) as $key) {
+    $rows[] = array(
+      $form[$key]['template_name']['#value'],
+      drupal_render($form[$key]['title']),
+      drupal_render($form[$key]['content_type']),
+      drupal_render($form[$key]['widget']),
+      drupal_render($form[$key]['visible']),
+    );
+  }
+
+  return theme('table', array('header' => $header, 'rows' => $rows)) . drupal_render_children($form);
+}
+
+/**
+ * Theme function to render backend lisy of widgets that require text trim.
+ */
+function theme__ding_nodelist_trim_text_table($variables) {
+  $header = array(
+    t('Widget'),
+    t('Title length'),
+    t('Teaser length'),
+  );
+
+  $rows = array();
+  $form = $variables['form'];
+  foreach (element_children($form) as $key) {
+    $rows[] = array(
+      $form[$key]['widget']['#value'],
+      drupal_render($form[$key]['title_trim']),
+      drupal_render($form[$key]['teaser_trim']),
+    );
+  }
+  return theme('table', array('header' => $header, 'rows' => $rows)) . drupal_render_children($form);
+}
+
+/**
+ * Ctools admin view pane's title.
+ */
+function ding_nodelist_content_type_admin_title($subtype, $conf, $context = NULL) {
+  if (!empty($conf['override_title_text'])) {
+    $output = t('Nodelist') . ': ' . filter_xss_admin($conf['override_title_text']);
+  }
+  else {
+    $output = t('Nodelist (untitled)');
+  }
+  return $output;
+}
+
+/**
+ * Callback to provide admin info (the preview in panels when building a panel).
+ */
+function ding_nodelist_content_type_admin_info($subtype, $conf, $context = NULL) {
+  // @todo: show details about nodelist configuration here.
+  $block = new stdClass();
+  $block->title = t('No information');
+  $info = '';
+  $block->content = $info;
+  return $block;
+}
+
+/**
+ * Run-time rendering of the body of the list.
+ *
+ * @param string $subtype
+ *   Content type sub-type.
+ * @param array $conf
+ *   Configuration as done at admin time.
+ * @param array $args
+ *   Special arguments to use.
+ * @param string $context
+ *   Possible context details.
+ *
+ * @return object
+ *   An object with at least title and content members.
+ */
+function ding_nodelist_content_type_render($subtype = '', $conf = array(), $args = array(), $context = '') {
+  $content_type = 'field_' . $conf['content_type'] . '_list_image';
```

Comment:
`$content_type` is an odd variable name for something that seems to contain the name of a specific field.